src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -175,9 +168,7 @@ internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, Diagnos
             var state = _state;
             var stateBuilder = ArrayBuilder<GeneratorState>.GetInstance(state.Generators.Length);
             var constantSourcesBuilder = ArrayBuilder<SyntaxTree>.GetInstance();
```

Comment:
Will the `SyntaxTree` instances returned from the `PostInit` phase particpate in caching? Don't know if we're caching at the `SyntaxTree` or content layer. Believe it's the former though hence we wouldn't get caching here. 

Likely not a big deal but think we should PROTOTYPE for future changes.

Reply:
Yes, we add them to the compilation before passing that to the driver state table builder, so they get included (the same object) each time, and should work with the cache as expected. I'll add a test to cover + confirm


---
In reply to: [632880939](https://github.com/dotnet/roslyn/pull/53383#discussion_r632880939) [](ancestors = 632880939)